pkg/envoy/server.go

```diff
@@ -1332,6 +1332,13 @@ func getPortNetworkPolicyRule(sel policy.CachedSelector, wildcard bool, l7Parser
 	if l7Rules.OriginatingTLS != nil {
 		r.UpstreamTlsContext = getCiliumTLSContext(l7Rules.OriginatingTLS)
 	}
+	if len(l7Rules.ServerNames) > 0 {
+		r.ServerNames = make([]string, 0, len(l7Rules.ServerNames))
+		for sni := range l7Rules.ServerNames {
```

Comment:
nit: you can fold this loop into `r.ServerNames = append(r.ServerNames, l7Rules.ServerNames...)`

Reply:
`l7Rules.ServerNames` is a map AFAIK a map can not be unpacked with `...`, will test though!

Reply:
Turns out map can not be unpacked with ellipses.

Reply:
I missed that it was a map. Thanks for checking though.